### PR TITLE
upgrade to gradle 7

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,79 +3,56 @@ plugins {
     id 'jacoco'
     id 'checkstyle'
     id 'pmd'
-    id 'nebula.lint' version '16.8.0'
-    id 'de.fuerstenau.buildconfig' version '1.1.8'
+    id 'nebula.lint' version '17.2.0'
+    id "com.github.gmazzo.buildconfig" version "3.0.3"
```
Comment (axthosarouris) [inline build.gradle:7]: the previous buildconfig file is not compatible with gradle 7

File: build.gradle [hunk 2]
```diff
@@ -3,79 +3,56 @@ plugins {
...
 apply plugin: 'idea'
 apply plugin: 'maven-publish'
 
-String versionNumber = '0.13.2'
-
 group 'com.github.bibsysdev'
-version versionNumber
+version '0.13.3'
+
+wrapper {
```
Comment (axthosarouris) [inline build.gradle:21]: maybe easier monitoring of the gradle wrapper? I can remove it if it is probematic

File: build.gradle [hunk 3]
```diff
@@ -3,79 +3,56 @@ plugins {
...
+    testImplementation libs.nva.commons.identifiers
 
-    compileOnly group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: jacksonVersion
-    testImplementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: jacksonVersion
+    compileOnly libs.bundles.jackson
+    testImplementation libs.bundles.jackson
 
-    compileOnly group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: jacksonVersion
-    testRuntimeOnly group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: jacksonVersion
+    testImplementation libs.bundles.unittesting
```
Comment (axthosarouris) [inline build.gradle:47]: bundles rule :)

File: build.gradle [hunk 4]
```diff
@@ -97,6 +74,11 @@ pmd {
     ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
     ruleSets = []
     ignoreFailures = false
+    pmdMain {
+        excludes = [
+                '**/BuildConfig.*'
```
Comment (axthosarouris) [inline build.gradle:79]: could not find a more generic way to exclude the buildconfig file. I would have liked to exclude all generated classes but could not find a way to do it. Patterns like `**/generated/**/* ` did not do the trick

File: build.gradle [hunk 5]
```diff
@@ -110,15 +92,16 @@ tasks.withType(Checkstyle) {
         html.enabled true
         html.stylesheet rootProject.resources.text.fromFile('config/checkstyle/checkstyle-simple.xsl')
     }
+    exclude '**/BuildConfig*'
```
Comment (axthosarouris) [inline build.gradle:95]: same

File: build.gradle [hunk 6]
```diff
@@ -110,15 +92,16 @@ tasks.withType(Checkstyle) {
...
-    appName = project.name
-    version = project.version
-    clsName = 'BuildConfig'
-    packageName = project.group
-    charset = 'UTF-8'
-    buildConfigField 'String', 'MODEL_VERSION', versionNumber
+    className("BuildConfig")   // forces the class name. Defaults to 'BuildConfig'
+    packageName(project.group)  // forces the package. Defaults to '${project.group}'
+    useJavaOutput() //forces the output to be Java
+    buildConfigField("String", "MODEL_VERSION", "\"${project.version}\"")
```
Comment (axthosarouris) [inline build.gradle:104]: kept only the field that is in use.

Comment (axthosarouris) [reply build.gradle:104]: The strange syntax is according to examples found in the plugin doc

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

**Pull request merged**